### ISSUE TITLE
SplitContainer - add percentage option for better scaling and UI responsiveness.

### ISSUE THREAD

**Issue by PanTrakX** [godotengine/godot-proposals#4033]
### Describe the project you are working on

On several GUI apps I have encountered this "issue".

### Describe the problem or limitation you are having in your project

The SplitContainer use pixel to set the split for the children. This works great if the game/application has the stretch mode set to 2d/viewport because the size of the window is kept the same when scaling it. If the stretch mode is set to disabled when scaling the window the size in pixels changes but the SplitContainer still keeps the children the same offset in pixel.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

My proposal is to have 2 options to set the split offset, one being in pixels and the other as percentage.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

On the SplitContainer have a dropdown option to either use 1)Pixel or 2)Percentage for the offset and if the percentage is selected then do the calculations that are needed.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think that yes, it can be worked around with a few lines of script

### Is there a reason why this should be core and not an add-on in the asset library?

This improves the UI responsiveness.

**Comment by Calinou** [comment]
> I think that yes, it can be worked around with a few lines of script

Indeed, you can listen to the `size_changed` signal on the parent control (if anchored) or the Viewport, and use that to resize the split offset.

**Comment by pycbouh** [comment]
> Example 1: If I want to have a HSplitContainer with 2 panels taking 50% each on a 1280x720 project display size I would set the pixel offset to 640. If I maximize the window on a 1920x1080 display then the first panel would take 33% and the second 66%.

Do you actually need the split functionality for your use case? I.e. do you need the ability for the user to change the split? Because if you only want to have two controls proportionally taking space of their container, then you should probably use a BoxContainer and configure size flag ratio for its two children. Ratio is basically a percent, so it will get you where you want.

**Comment by PanTrakX** [comment]
> Do you actually need the split functionality for your use case? I.e. do you need the ability for the user to change the split?

Yes. 
In the following video you can see how it works compared to blender's areas (which is what I tried to recreate in this project). 
I can implement such functionality myself on the projects that needs this but I proposed it here cause I think that it would be a helpful addition in the engine because this is a feature found in many modern apps. 

Also something else that I want to mention (that has been already proposed) is to be able to have smaller gap (like in blender) or an almost no gap (like in VSCode) and without the need to use a sprite for the grabber

https://user-images.githubusercontent.com/62219596/154817547-10700071-2c5c-4139-a5d9-fed39c284d3e.mp4

**Comment by Calinou** [comment]
> Also something else that I want to mention (that has been already proposed) is to be able to have smaller gap (like in blender) or an almost no gap (like in VSCode) and without the need to use a sprite for the grabber

This is already being tracked in https://github.com/godotengine/godot-proposals/issues/3635.

**Comment by Remixful** [comment]
I can't figure out how to work with the split container for the life of me. I really hope we can get a percentage option instead. The offset option is unintuitive, and I'm upset with whoever thought it was okay. 😭

**Comment by Godaux** [comment]
> I can't figure out how to work with the split container for the life of me. I really hope we can get a percentage option instead. The offset option is unintuitive, and I'm upset with whoever thought it was okay. 😭

If it helps anyone else I'm currently just using the **resized** signal on the SplitContainer to scale the **split_offset** with the same factor as current SplitContainer.size.x / original SplitContainer.size.x
